### PR TITLE
Add unit tests for alluxio utils

### LINKED ISSUE #6597: Alluxio unit testing
**Is your feature request related to a problem? Please describe.**
Currently we don't have any internal support for testing alluxio, either unit or integratoin testing. We have another issue to add integration testing using real alluxio system, but it would also be nice to setup a bit of framework to do unit testing.  To do this we will need to stub out some things that expect to get alluxio master etc and ideally have some way to mock the readers to be able to reach in to see we are replacing the paths properly.

### PR DESCRIPTION
**Pull request by res-life**
Closes #6597

Add unit cases for methods of `AlluxioUtils`:
- updateFilesTaskTimeIfAlluxio
- checkIfNeedsReplaced
- autoMountIfNeeded

Note: Do not add cases for the convert-time algorithm because it's difficult to mock the behaviors of `InMemoryFileIndex`

Signed-off-by: Chong Gao <res_life@163.com>

### CODE REVIEW

**Comment by res-life** [comment]
build

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/AlluxioMasterAndPortReader.scala [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      if (alluxio_master == null) {
+        throw new RuntimeException(
+          s"Can't find alluxio.master.hostname from $homePath/conf/alluxio-site.properties.")
+      }
+
+      val alluxio_port = prop.getProperty("alluxio.master.rpc.port", "19998")
+      (alluxio_master, alluxio_port)
+    } catch {
+      case _: FileNotFoundException =>
+        throw new RuntimeException(s"Not found Alluxio config in " +
```
Comment (tgravescs) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/AlluxioMasterAndPortReader.scala:54]: can we reword to be "Alluxio config file not found in" ...

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/AlluxioFS.scala [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  private var masterHost: Option[String] = None
+  private var masterPort: Option[Int] = None
+  private var alluxioUser: String = ""
+  private var s3AccessKey: Option[String] = None
+  private var s3SecretKey: Option[String] = None
+
+  def setHostAndPort(masterHost: Option[String], masterPort: Option[Int]): AlluxioFS = {
+    this.masterHost = masterHost
+    this.masterPort = masterPort
+    this
```
Comment (tgravescs) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/AlluxioFS.scala:41]: why do we need to return this here?  You have an instance so just set it

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/AlluxioUtils.scala [hunk 3]
```diff
@@ -87,10 +79,11 @@ object AlluxioUtils extends Logging with Arm {
   private var alluxioMasterPort: Option[Int] = None
```
Comment (tgravescs) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/AlluxioUtils.scala:79]: why do we still have this if its in the AlluxioFS?  Couldn't we change to get from there and you could mock it wihtout reset?

**Review by tgravescs — COMMENTED** [review]
this is a great start but it doesn't close https://github.com/NVIDIA/spark-rapids/issues/6597
I would really like to see some tests that make sure the parquet code does the right things as well. That is the complicated part that I could see be easily broken if things get changed.  We can file a separate one if needed but would like to test the combinations of:

- replacement mode (automount vs specific replace), convert time vs task time, parquet reader type (perfile, multithreaded, coalescing)

We should ensure that each of those combinations replaces the paths when it should

**Comment by res-life** [comment]
build

**Comment by res-life** [comment]
build

**Comment by res-life** [comment]
build

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/AlluxioUtils.scala (hunk 3 above)
Comment (res-life) [reply sql-plugin/src/main/scala/com/nvidia/spark/rapids/AlluxioUtils.scala]: Done.

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/AlluxioFS.scala (hunk 2 above)
Comment (res-life) [reply sql-plugin/src/main/scala/com/nvidia/spark/rapids/AlluxioFS.scala]: Done.

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/AlluxioMasterAndPortReader.scala (hunk 1 above)
Comment (res-life) [reply sql-plugin/src/main/scala/com/nvidia/spark/rapids/AlluxioMasterAndPortReader.scala]: Done.

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/AlluxioMasterAndPortReader.scala [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+ */
+class AlluxioMasterAndPortReader {
+
+  private val alluxioHome: String = "/opt/alluxio-2.8.0"
+
+  def readAlluxioMasterAndPort(): (String, String) = {
+    readAlluxioMasterAndPort(alluxioHome)
+  }
+
+  // Default to read from /opt/alluxio-2.8.0 if not setting ALLUXIO_HOME
```
[the remark this answers is not in the document]
Comment (res-life) [reply sql-plugin/src/main/scala/com/nvidia/spark/rapids/AlluxioMasterAndPortReader.scala]: Done.

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/AlluxioMasterAndPortReader.scala [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+/**
+ * Alluxio master address and port reader.
+ * It reads from `/opt/alluxio-2.8.0/conf/alluxio-site.properties`
+ */
+class AlluxioMasterAndPortReader {
+
+  private val alluxioHome: String = "/opt/alluxio-2.8.0"
+
+  def readAlluxioMasterAndPort(): (String, String) = {
```
[the remark this answers is not in the document]
Comment (res-life) [reply sql-plugin/src/main/scala/com/nvidia/spark/rapids/AlluxioMasterAndPortReader.scala]: Done.

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/AlluxioMasterAndPortReader.scala [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+import java.io.FileNotFoundException
+import java.util.Properties
+
+import scala.io.BufferedSource
+
+/**
+ * Alluxio master address and port reader.
+ * It reads from `/opt/alluxio-2.8.0/conf/alluxio-site.properties`
+ */
+class AlluxioMasterAndPortReader {
```
[the remark this answers is not in the document]
Comment (res-life) [reply sql-plugin/src/main/scala/com/nvidia/spark/rapids/AlluxioMasterAndPortReader.scala]: Done.

**Comment by res-life** [comment]
Created a follow-up issue to add more test cases:
https://github.com/NVIDIA/spark-rapids/issues/7240

**Review by tgravescs — APPROVED** [review]

**Pull request merged**